### PR TITLE
S3949: Implement ShouldExecute for VB.NET

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Part of #7147

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs [hunk 1]
```diff
@@ -29,6 +29,28 @@ public sealed class CalculationsShouldNotOverflow : CalculationsShouldNotOverflo
...
+        {
+            if (!Result)
+            {
+                Result = node.IsAnyKind(
+                    SyntaxKind.AddExpression,
+                    SyntaxKind.AddAssignmentStatement,
+                    SyntaxKind.MultiplyExpression,
+                    SyntaxKind.MultiplyAssignmentStatement,
+                    SyntaxKind.SubtractExpression,
+                    SyntaxKind.SubtractAssignmentStatement);
```
Comment (Tim-Pohlmann) [inline analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs:51]:
```suggestion
                Result = node.IsAnyKind(
                    SyntaxKind.AddExpression,
                    SyntaxKind.AddAssignmentStatement,
                    SyntaxKind.SubtractExpression,
                    SyntaxKind.SubtractAssignmentStatement,
                    SyntaxKind.MultiplyExpression,
                    SyntaxKind.MultiplyAssignmentStatement,
                    SyntaxKind.ExponentiateExpression,
                    SyntaxKind.ExponentiateAssignmentStatement);
```

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM, just add `ExponentiateExpression`.

File: analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs:51]: `a ^ b` returns `Double` and cannot overflow => no need to run

Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs:51]: Should we update the rule implementation then?

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs:51]: Hmm, yes, indeed.

**Pull request merged**